examples/run/cpu/seq_example.cpp

```diff
@@ -28,46 +29,69 @@ int seq_run(const std::string& detector_file, const std::string& cells_dir,
     uint64_t n_modules = 0;
     uint64_t n_measurements = 0;
     uint64_t n_spacepoints = 0;
+    uint64_t n_internal_spacepoints = 0;
+    uint64_t n_seeds = 0;
 
     // Memory resource used by the EDM.
     vecmem::host_memory_resource host_mr;
 
     traccc::clusterization_algorithm ca;
-    traccc::seeding_algorithm sa(&host_mr);
+    traccc::seeding_algorithm sa(host_mr);
+    traccc::track_params_estimation_algorithm tp;
 
     // Loop over events
     for (unsigned int event = 0; event < events; ++event) {
 
         // Read the cells from the relevant event file
-        std::string io_cells_file =
-            traccc::data_directory() + cells_dir + "/" +
-            traccc::get_event_filename(event, "-cells.csv");
-        traccc::cell_reader creader(
-            io_cells_file, {"geometry_id", "hit_id", "cannel0", "channel1",
-                            "activation", "time"});
         traccc::host_cell_container cells_per_event =
-            traccc::read_cells(creader, host_mr, &surface_transforms);
+            traccc::read_cells_from_event(event, cells_dir, surface_transforms,
+                                          host_mr);
 
         /*-------------------
             Clusterization
           -------------------*/
 
-        auto ca_result = ca(cells_per_event);
-        auto& measurements_per_event = ca_result.first;
-        auto& spacepoints_per_event = ca_result.second;
+        traccc::clusterization_algorithm::input_type ca_input(cells_per_event);
+        traccc::clusterization_algorithm::output_type ca_output;
+
+        ca_output = ca(ca_input);
+
+        auto& measurements_per_event = ca_output.first;
+        auto& spacepoints_per_event = ca_output.second;
+
+        /*-----------------------
+             Seeding algorithm
+          -----------------------*/
+
+        traccc::seeding_algorithm::input_type sa_input(spacepoints_per_event);
+        traccc::seeding_algorithm::output_type sa_output;
+
+        sa_output = sa(sa_input);
+
+        auto& internal_sp_per_event = sa_output.first;
+        auto& seeds = sa_output.second;
+
+        /*----------------------------
+          Track params estimation
+          ----------------------------*/
+
+        traccc::track_params_estimation_algorithm::input_type tp_input(seeds);
+        traccc::track_params_estimation_algorithm::output_type tp_output;
+
+        tp_output = tp(tp_input);
+
+        auto& params = tp_output;
```

Comment:
With the updated algorithms semantics this could surely be simplified quite a bit. Most of these lines should not be needed anymore.

Reply:
Done